dev/core/src/com/google/gwt/core/linker/SymbolMapsLinker.java

```diff
@@ -335,6 +358,70 @@ public Object merge(String extKey, Object oldVal, Object newVal) {
     return artifacts;
   }
 
+  private static void embedSourcesInSourceMaps(TreeLogger logger, StandardLinkerContext context,
+                                               ArtifactSet artifacts,
+                                               SourceMapGeneratorV3 sourceMapGenerator,
+                                               int totalPrefixLines, String sourceMapString,
+                                               String partialPath)
+      throws SourceMapParseException {
+    sourceMapGenerator.setStartingPosition(totalPrefixLines, 0);
+    SourceMapConsumerV3 section = new SourceMapConsumerV3();
+    section.parse(sourceMapString);
+    section.visitMappings(sourceMapGenerator::addMapping);
+
+    Iterator extensions = section.getExtensions().entrySet().iterator();
+
+    while (extensions.hasNext()) {
+      Entry<String, Object> entry = (Entry) extensions.next();
+      String extensionKey = (String) entry.getKey();
+      sourceMapGenerator.addExtension(extensionKey, entry.getValue());
+    }
+
+    ResourceLoader resourceLoader = ResourceLoaders.fromContextClassLoader();
+
+    Map<String, EmittedArtifact> generatedSources = new java.util.HashMap<>();
+    artifacts.find(EmittedArtifact.class)
+        .forEach(emittedArtifact -> {
+            if (Visibility.Source == emittedArtifact.getVisibility()) {
+              generatedSources.put(emittedArtifact.getPartialPath(), emittedArtifact);
+            }
+        });
+
+    for (String sourceFileName : section.getOriginalSources()) {
+      String content;
+      try {
+        InputStream cis = loadSource(logger, sourceFileName, generatedSources,
+            resourceLoader);
```

Comment:
Should be try-with-resources, right, so we are sure to close the file?

Reply:
The line after that is a check for a null and try to use a difference resource for loading the content, so I guess using the try with resource wont work here, but we can add a finally block to ensure we close the InputStream.